ts/src/test/Exchange/test.fetchOHLCV.ts

```diff
@@ -11,16 +12,54 @@ async function testFetchOHLCV (exchange, skippedProperties, symbol) {
     if (!exchange.inArray (chosenTimeframeKey, timeframeKeys)) {
         chosenTimeframeKey = timeframeKeys[0];
     }
-    const limit = 10;
-    const duration = exchange.parseTimeframe (chosenTimeframeKey);
-    const since = exchange.milliseconds () - duration * limit * 1000 - 1000;
+    const limit = 50;
+    const durationMs = exchange.parseTimeframe (chosenTimeframeKey) * 1000;
+    const since = exchange.milliseconds () - durationMs * limit - 1000;
     const ohlcvs = await exchange.fetchOHLCV (symbol, chosenTimeframeKey, since, limit);
     assert (Array.isArray (ohlcvs), exchange.id + ' ' + method + ' must return an array, returned ' + exchange.json (ohlcvs));
     const now = exchange.milliseconds ();
     for (let i = 0; i < ohlcvs.length; i++) {
         testOHLCV (exchange, skippedProperties, method, ohlcvs[i], symbol, now);
     }
-    // todo: sorted timestamps check
+    const logText = testSharedMethods.logTemplate (exchange, method, ohlcvs);
+    // check boundaries
+    const barsLength = ohlcvs.length;
+    const barsLengthAboveZero = barsLength > 0;
+    // ensure bars amount is less then limit
+    assert (barsLength <= limit, 'Returned bars amount (' + barsLength.toString () + ') is more than requested (' + limit.toString () + ')' + logText);
+    // ensure bars amount is more than zero
+    if (!('compareAmountToZero' in skippedProperties)) {
+        assert (barsLengthAboveZero, 'Returned bars amount should be more then zero' + logText);
+    }
+    // if no bars were returned, we can't do any more checks
+    if (!barsLengthAboveZero) {
+        return;
+    }
+    const minActrualTs = ohlcvs[0][0];
+    const maxActualTs = ohlcvs[barsLength - 1][0];
+    const maxExpectedTs = minActrualTs + durationMs * limit;
+    // ensure that timestamps are greaterOrEqual than since
```

Comment:
@ttodua  let's say the API returns 3 months maximum of candles and we request 4 months old data, it is expected that the since < firstTs no?

Reply:
you are just trying to name the theoretical cases that might fail, and I do not disagree on those cases, but what I emphasize is the actual tests we do:
- if we do tests for minute data, we are doing tests for last 100 (up to maximum 1000 bars), just to "test" the ohlcv structure and it's basic functionality (o/h/l/c/v values etc...) and even 1000 one-minute bars does not exceed last 2 days. all exchanges would provide last 2 days of diapason
- if we test for daily bars, we would also test last i.e. 100 bars for daily candles. all exchagnes would provide 3 months for daily bars. if you find any exchange which does not, we can add such crappy exceptional exchange to skips.

so, I don't see any issue you are describing - we nowhere have such tests for fetchOHLCV where we are doing i.e. 3months of minute data fetchings... 
so, we have not failed tests even when these changes will be merged and I dont see a problem actually. if we will ever have more complex tests added for fetchOHLCV i will ofc re-evaluate those tests.